test/src/sdk-harness/test_projects/ec_recover/mod.rs

```diff
@@ -0,0 +1,116 @@
+> {
+    let mut rng = StdRng::seed_from_u64(1000);
+    let msg_bytes: Bytes32 = rng.gen();
+    let private_key = SecretKey::random(&mut rng);
+    let public_key = PublicKey::from(&private_key);
+    let msg = unsafe { Message::from_bytes_unchecked(*msg_bytes) };
+    let sig = Signature::sign(&private_key, &msg);
+    let sig_bytes: Bytes64 = Bytes64::try_from(sig).unwrap();
+    let mut wallet = WalletUnlocked::new_from_private_key(private_key, None);
+
+    let num_assets = 1;
+    let coins_per_asset = 10;
+    let amount_per_coin = 15;
+    let (coins, _asset_ids) = setup_multiple_assets_coins(
+        wallet.address(),
+        num_assets,
+        coins_per_asset,
+        amount_per_coin,
+    );
+    let (provider, _socket_addr) = setup_test_provider(coins.clone(), vec![], None).await;
+    wallet.set_provider(provider);
+
+    let contract_id = Contract::deploy(
+        "test_projects/ec_recover/out/debug/ec_recover.bin",
+        &wallet,
+        TxParameters::default(),
+        StorageConfiguration::default(),
+    )
+    .await?;
+
+    let contract_instance = EcRecoverContract::new(contract_id, wallet.clone());
+
+    Ok((
+        contract_instance,
+        private_key,
+        public_key,
+        wallet,
+        msg,
+        sig_bytes,
+    ))
+}
+
+#[tokio::test]
+async fn can_recover_public_key() {
+    let (contract, _secret, public_key, _wallet, msg, sig_bytes) = setup_env().await.unwrap();
+    let sig_r = &sig_bytes[..32];
+    let sig_v_s = &sig_bytes[32..];
+    let response = contract
+        .methods()
+        .recover_pub_key(
+            Bits256(sig_r.try_into().unwrap()),
+            Bits256(sig_v_s.try_into().unwrap()),
+            Bits256(msg.into()),
+        )
+        .call()
+        .await
+        .unwrap();
+
+    let first = response.value.0;
+    let second = response.value.1;
+    let arrays: [[u8; 32]; 2] = [first.0, second.0];
+    let joined: Vec<u8> = arrays.into_iter().flat_map(|s| s.into_iter()).collect();
+    let joined_array: [u8; 64] = joined.try_into().unwrap();
+    let pubkey = Bytes64::new(joined_array);
+
+    assert_eq!(pubkey, Bytes64::new(*public_key));
+}
+
+#[tokio::test]
+async fn can_recover_address() {
+    let (contract, _secret, _public_key, wallet, msg, sig_bytes) = setup_env().await.unwrap();
+    let sig_r = &sig_bytes[..32];
+    let sig_v_s = &sig_bytes[32..];
+    let response = contract
+        .methods()
+        .recover_address(
+            Bits256(sig_r.try_into().unwrap()),
+            Bits256(sig_v_s.try_into().unwrap()),
+            Bits256(*msg),
+        )
+        .call()
+        .await
+        .unwrap();
+
+    // let first = response.value.0;
+    // let second = response.value.1;
+    // let arrays: [[u8; 32]; 2] = [first.0, second.0];
+    // let joined: Vec<u8> = arrays.into_iter().flat_map(|s| s.into_iter()).collect();
+    // let joined_array: [u8; 64] = joined.try_into().unwrap();
+    // let pubkey = Bytes64::new(joined_array);
```

Comment:
Are these intentionally left here?

Reply:
hmm, nope. Those look old, maybe I forgot to push my latest change.
I'll delete the old E2E test now. The only reason I've left test-sig-gen-util for now it that it's still kind of linked to the evm ec recover test which is the last one to be converted.